omnibus/files/private-chef-cookbooks/private-chef/libraries/elasticsearch_index_provider.rb

```diff
@@ -21,14 +21,21 @@ class ElasticSearchIndex < Chef::Provider::LWRPBase
         end
       end
 
-      def index_exists?
+      ## We check if the elasticsearch index exists before creating it.
+      ## But Elasticsearch returns 404 for a short duration during startup before it stabilizes and can return a 200.
+      ## As a result, we incorrectly try to recreate an index when it actually exists in the system.
+      ## Re-creating an existing index is a 400 "Bad Request" error.
+      def retry_index_exists?(count)
         solr_server.get("/#{new_resource.index_name}")
         true
       rescue Net::HTTPClientException => e
-        if e.response && e.response.code == '404'
-          false
+        raise unless (e.response && e.response.code) == '404'
```

Comment:
Should this be something like: 

```suggestion
        raise unless (e.response && e.response.code == '404')
```